File: server/src/data/repositories/EmployeeRepository.ts
import { db } from "@data/db";
import { employee } from "@data/db/schema";
import { Employee } from "@domain/entities/Employee";
import type { IEmployeeRepository } from "@domain/repositories/IEmployeeRepository";

export class EmployeeRepository implements IEmployeeRepository {
  async createEmployee(
    employeeData: Employee
  ): Promise<{ employee: Employee }> {
    const { name, estimatedHours, squadId } = employeeData;

    const [data] = await db
      .insert(employee)
      .values({
        name,
        estimatedHours,
        squadId,
      })
      .returning();

    return {
      employee: new Employee(data),
    };
  }

  async getEmployees(): Promise<Employee[]> {
    const employees = await db.select().from(employee);

    return employees.map((employee) => new Employee(employee));
  }
}

File: ui/lib/features/domain/repositories/employee_repository.dart
import 'package:hours_control/features/domain/entities/employee_entity.dart';

abstract class EmployeeRepository {
  Future<List<EmployeeEntity>> fetchEmployees();

  Future<EmployeeEntity> createEmployee({
    required String name,
    required int estimatedHours,
    required int squadId,
  });
}
